boards/arm/adafruit_itsybitsy_m4_express/adafruit_itsybitsy_m4_express.dts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2020 Google LLC.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <atmel/samd5xx19.dtsi>
+
+/ {
+	model = "Adafruit ItsyBitsy M4 Express";
+	compatible = "adafruit,itsybitsy-m4-express";
+
+	chosen {
+		zephyr,console = &sercom3;
+		zephyr,shell-uart = &sercom3;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,code-partition = &code_partition;
+	};
+
+	/* These aliases are provided for compatibility with samples */
+	aliases {
+		led0 = &led0;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		led0: led_0 {
+			gpios = <&porta 22 0>;
+			label = "LED";
+		};
+	};
+};
+
+&cpu0 {
+	clock-frequency = <120000000>;
+};
+
+&sercom3 {
+	status = "okay";
+	compatible = "atmel,sam0-uart";
+	current-speed = <115200>;
+	rxpo = <1>;
+	txpo = <0>;
+};
+
+&flash0 {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		boot_partition: partition@0 {
+			label = "uf2";
+			reg = <0x00000000 DT_SIZE_K(16)>;
+			read-only;
+		};
+
+		code_partition: partition@4000 {
+			label = "code";
+			reg = <DT_SIZE_K(16) DT_SIZE_K(512-8-16)>;
```

Comment:
Could you please check if this is correct?
If you reserve 16kB for Bootloader and 16kB fot storage then the code should be 512 - 32, unless I am missing something.

Also, it is sort of a style convention to use "absolute" addresses to indicate the reg addr, instead of size macros.

Reply:
Good spotting - that's a copy-paste error from the SAMD21 which has an 8k bootloader.

Fixed.